Added a `Receive` page to `Daily Spending Wallet` (#878)
BitcoinDesign/Guide: guide/daily-spending-wallet/requesting/receiving.md

## Receiving on-chain payments

To receive on-chain payments and maintain a single lightning balance a [lightning wallet server (LWS)]({{ '/guide/how-it-works/lightning-services/#what-are-lightning-wallet-servers'| relative_url }}) is required to conduct [submarine swaps](https://blog.muun.com/a-closer-look-at-submarine-swaps-in-the-lightning-network/) on the users behalf. Submarine swaps move bitcoin between the bitcoin and lightning networks.

{% include tip/open.html color="blue" icon="info" label="Experimental options" %}

Experimental options such as [Peerswap](https://www.peerswap.dev/) and [Collaborative funding](https://bitcoinops.org/en/topics/dual-funding/) may be more trustless options that could be used to receive on-chain payments whilst maintaining a single lightning balance.

{% include tip/close.html %}

### Refunds

> There may be scenarios where a submarine swap is not completed successfully, such as when the wallets LSP is offline. In these cases, the users will need to claim the on-chain funds that failed to be swapped by sending them to a on-chain address. 

Reviewer: > users will need to claim the on-chain funds that failed to be swapped by sending them to a on-chain address. 

Can you elaborate a bit more on this part? I am not sure it's clear enough.

Reviewer 2: I assume the LSP will try this multiple times before going this refund route?

Author: Will clarify this, yes they can. The LWS is actually who does the swap* Need to fix that. The LSP opens a channel though if it is required. 

Some context on how Breez deals with these refunds: https://doc.breez.technology/Adding-Funds-via-Submarine-Swaps.html?highlight=refund#how-can-users-claim-their-refunds

   alt-text = "Screen showing a warning error notification prompting the user to claim a refund of their failed submarine swap."
   width = 250
   height = 541
   modalWidth = 250
   modalHeight = 541
%}

{% include picture.html
   image = "/assets/images/guide/daily-spending-wallet/receiving/RefundActivity.png"
   retina = "/assets/images/guide/daily-spending-wallet/receiving/RefundActivity@2x.png"
   modalImage = "/assets/images/guide/daily-spending-wallet/receiving/RefundActivity@2x.png"
   layout = "shadow"
   caption = "The activity page is another place where you can indicate to a user that they need to claim their refund."
   alt-text = "Screen showing the activity menu with an entry for the user to claim their refunded payment."
   width = 250
   height = 541
   modalWidth = 250
   modalHeight = 541
%}

{% include picture.html
   image = "/assets/images/guide/daily-spending-wallet/receiving/ClaimRefund.png"
   retina = "/assets/images/guide/daily-spending-wallet/receiving/ClaimRefund@2x.png"
   modalImage = "/assets/images/guide/daily-spending-wallet/receiving/ClaimRefund@2x.png"
   layout = "shadow"

>    caption = "Make it simple for a user to claim their refund."

Author: They have to, how else would they claim it? Technically these funds are on the users local on-chain wallet (which is hidden away and only used for this purpose). If the swap fails it can't be tried again for this reason. 

They may not be though, they could be sending back to the person that tried to pay them via the swap (will clarify this)

>    caption = "Make it simple for a user to claim their refund."

Author: Retracting that, the funds are held by the LWS if the swap fails (custodial risk, a good reason to move to PeerSwap when its ready).

>    caption = "Make it simple for a user to claim their refund."

Author: Taking heavily from [these]( https://docs.lightning.engineering/the-lightning-network/multihop-payments/understanding-submarine-swaps) docs and how Breez deals with this.

> So it seems like this could just be an on-chain address that belongs to the local on-chain wallet.

_I think_ how Breez and such does it is that the smart contract generated is between the Breez user and the Breez LWS. Based on the docs above the sender (which is most likely a third party) and receiver (Breez user) generate the contract and if it fails the sender claims the refund from the contract by signing it which isn't the case in Breez / this example as the user (receiver) is claiming the refund. 

I'm assuming Breez uses the users local on-chain wallet like you mentioned and payment goes from sender -> Breez user -> smart contract is constructed between Breez user and Breez LWS -> If it fails user can 'sweep' these on-chain funds from the contract to an on-chain address (whether they control it or not). 

> Retry the LWS automatically later?

I think this could be done. Technically it would just be the user 'claiming' the on-chain funds from the swap contract directly into a new contract. This should not be automatically though as it costs on-chain fees. User should click something to do this and be shown the fees. Another thing I didn't add is this can't be claimed until the time lock on the contract expires (usually 48 hours). 

> If the amount is high enough, simply open a new channel to the LSP?

Unless you used collaborative funding this channel would only have outbound capacity though making it not so useful.

> On the other hand, I think "refund" could be confusing from a user perspective. Like, what am I getting refund? Did a return something at a store?

Well it depends, if the payment is from a third party (not the receiver themselves) you could refund them the failed payment. You could keep their funds by sending to an on-chain wallet you control too but this requires the user to have one of those setup. If you are just sending it to yourself (say if you are just funding your Breez wallet) then it isn't really a refund it's just returning the funds back to where they came from. The app wouldn't be able to know the difference between these though. 

Rather than say refund we could just have the warning notification, the user clicks it and it says "Your incoming payment was not successful, you can re-try it in x amount of time." The x time is the time lock which needs to expire before it can be claimed. 

For advanced users or if the user needs to actually refund the payment to a third party for some reason there can be option to claim to an external on-chain address (not re-try the swap)

>    caption = "Make it simple for a user to claim their refund."

Reviewer 3: I have a TODO from our call yesterday to help with ideas for replacing "refund", so here goes.

> Unless you used collaborative funding this channel would only have outbound capacity though making it not so useful.

I actually disagree with that. The channel is very useful -- they can use it to send bitcoin. They can still receive off the channel from First Use.

And consider one of your statements from the page, that a failed swap could be caused by: "receiving more than the users inbound capacity". So that means this swap could actually be a sizable chunk of money, which they may presumably want to use to pay for some things with later. So the channel would come in handy.

Anyways, on to my suggestions. This is rough and not pretty sounding, just thinking about a good way to frame this content. Maybe something like this:

## ~~Refunds~~

~~Refunds are required in some scenarios when a submarine swap can not be completed successfully. A refund is when a swap fails and the incoming on-chain bitcoin is now in control by the LWS. Users will need to claim these on-chain funds as a refund.~~

~~Below are examples of when a refund will be triggered:~~

- ~~LWS swap service is offline~~
- ~~Receiving more than the users inbound capacity~~
- ~~Re-using a swap-address to receive an on-chain payment~~

## Handling on-chain funds

In the vast majority of scenarios, the submarine swap succeeds and the user is able to receive on-chain bitcoin to their lightning wallet without even thinking about this complex process happening behind the scenes. However, in some edge cases, the swap can fail.

- LWS swap service is offline
- Receiving more than the users inbound capacity
- Re-using a swap-address to receive an on-chain payment

In those situations, this wallet will attempt to automatically retry the submarine swap later. The funds will show up as "pending" in the user's Activity. However, if the swap fails repeatedly, then it can turn this into a new channel for the user. The wallet prompts the user if they are OK with the channel open fee. If the user declines, then the wallet simply has on-chain funds. This transaction is highlighted *[differentiated in some way]* in the Activity list. The user can tap it and send the funds to another on-chain wallet.

   alt-text = "Screen showing a form to claim the refund to an on-chain address."
   width = 250
   height = 541
   modalWidth = 250
   modalHeight = 541
%}

</div>

## Received payment

> Once a payment is received it should be clearly indicated to the user and their balance should be updated. Ensure fees charged, if any, are clear to the user when viewing the payments details. Find more information on presenting a payment history on our [activity]({{ '/guide/daily-spending-wallet/activity/'| relative_url }}) page. 

Reviewer 2: ```suggestion
Received payment are clearly indicated to the user, and their balance is updated. Charged fees, if any, are clear to the user when viewing the payments details. Find more information on presenting a payment history on our [activity]({{ '/guide/daily-spending-wallet/activity/'| relative_url }}) page. 
```

> The option to save and share a receipt should be readily available.

Reviewer: I don't know what readily available means, just available maybe?

Author: Just means easily, but agree it's not necessary will remove

> The option to save and share a receipt should be readily available.

Reviewer 2: ```suggestion
The option to save and share a receipt is readily available.
```

## Receiving on-chain payments

This wallet maintains a single lightning balance. To do this but still allow on-chain sending and requesting, a LWS is used that conducts [submarine swaps](https://blog.muun.com/a-closer-look-at-submarine-swaps-in-the-lightning-network/) between lightning and on-chain.

{% include tip/open.html color="blue" icon="info" label="Experimental options" %}

[Peerswap](https://www.peerswap.dev/) or [Collaborative funding](https://bitcoinops.org/en/topics/dual-funding/) may be more trustless options that do not require a third-party LWS to receive and send on-chain payments. Though these are not currently widely adopted.

{% include tip/close.html %}

### Refunds

Refunds are required in some scenarios when a submarine swap can not be completed successfully. A refund is when a swap fails and the incoming on-chain bitcoin is now in control by the LWS. Users will need to claim these on-chain funds as a refund.

> Reasons for refunds will vary based on how the LWS swap service is implemented but below are some common scenarios:

Author: Good point, will adjust.

   modalHeight = 542
%}

{% include picture.html
   image = "/assets/images/guide/daily-spending-wallet/receiving/RefundActivity.png"
   retina = "/assets/images/guide/daily-spending-wallet/receiving/RefundActivity@2x.png"
   modalImage = "/assets/images/guide/daily-spending-wallet/receiving/RefundActivity@2x.png"
   layout = "shadow"
   caption = "The activity page is another place where you can indicate to a user that they need to claim their refund."
   alt-text = "Screen showing the activity menu with an entry for the user to claim their refunded payment."
   width = 250
   height = 542
   modalWidth = 250
   modalHeight = 542
%}

{% include picture.html
   image = "/assets/images/guide/daily-spending-wallet/receiving/ClaimRefund.png"
   retina = "/assets/images/guide/daily-spending-wallet/receiving/ClaimRefund@2x.png"
   modalImage = "/assets/images/guide/daily-spending-wallet/receiving/ClaimRefund@2x.png"
   layout = "shadow"
   caption = "Make it simple for a user to claim their refund."
   alt-text = "Screen showing a form to claim the refund to an on-chain address."
   width = 250
   height = 542
   modalWidth = 250
   modalHeight = 542
%}

</div>

## Received payment

> Received payments to this wallet are clearly indicated to the user. Fees paid, if any, are clear to the user when viewing the payments details. Users can save and share a receipt.

Reviewer 3: ```suggestion
When this wallet receives a payment, it notifies the user. If any fees are paid, they are clear to the user when viewing the payment details. Users can save and share a receipt.
```
I find the wording to be a little redundant. "Clear to the user" is used twice in a row.

More information on presenting a payment history on our [activity]({{ '/guide/daily-spending-wallet/activity/'| relative_url }}) page.

<div class="image-slide-gallery">

{% include picture.html
   image = "/assets/images/guide/daily-spending-wallet/receiving/received.png"
   retina = "/assets/images/guide/daily-spending-wallet/receiving/received@2x.png"
   modalImage = "/assets/images/guide/daily-spending-wallet/receiving/received@2x.png"
   layout = "shadow"
   caption = "Notify the user once their incoming payment is received."
   alt-text = "Screen showing a payment has been received."
   width = 250
   height = 542
   modalWidth = 250
   modalHeight = 542
%}

{% include picture.html
   image = "/assets/images/guide/daily-spending-wallet/receiving/ReceivedPaymentActivity.png"
   retina = "/assets/images/guide/daily-spending-wallet/receiving/ReceivedPaymentActivity@2x.png"
   modalImage = "/assets/images/guide/daily-spending-wallet/receiving/ReceivedPaymentActivity@2x.png"
   layout = "shadow"

>    caption = "Make it clear if extra fees were incurred, such as from a new channel open, to the user."

Reviewer 3: ```suggestion
   caption = "Make it clear to the user if extra fees were incurred, such as from a new channel open."
```
Simplify the way it reads with a slight re-arrangement

   modalImage = "/assets/images/guide/daily-spending-wallet/receiving/ReceiveOfflineNotification@2x.png"
   layout = "float-left-desktop -background -shadow"
   caption = "Notify the user that they have a payment incoming and that they should open their app to receive it."
   alt-text = "Screen showing a notification that is asking the user to open their wallet so they can receive their payment."
   width = 250
   height = 542
   modalWidth = 250
   modalHeight = 542
%}

This wallet uses a [lightning wallet server (LWS)]({{ 'guide/how-it-works/lightning-services/#what-are-lightning-wallet-servers' | relative_url }}) to intercept and hold payments for offline users until they come online. The LWS sends a push notification to the user to open their wallet so the payment can be received.

To avoid payment failures, when generating a request, it's indicated to the user that they should keep their wallet open until the payment is received. More on this [here]({{ '/guide/daily-spending-wallet/requesting/#creating-the-request' | relative_url }}).

</div>

{% include tip/open.html color="blue" icon="info" label="Offline payments" %}

> Offline payments, also known as async payments, are still not standards in the lightning ecosystem. Keep this in mind when developing your product. More on this on a [lightning services page]({{ '/guide/how-it-works/lightning-services/#receive-payments-offline' | relative_url }}).

Reviewer 2: ```suggestion
Offline payments, also known as async payments, are still not standards in the lightning ecosystem. Keep this in mind when developing your product. More on this on the [lightning services page]({{ '/guide/how-it-works/lightning-services/#receive-payments-offline' | relative_url }}).
```

{% include tip/close.html %}

## Receiving lightning payments 

Once the user is online, a payment channel with [inbound liquidity]({{ '/guide/how-it-works/liquidity/'| relative_url }}) is required to receive a payment. If the user already has a payment channel with enough inbound liquidity, the payment will be [received]({{ '/guide/daily-spending-wallet/requesting/receiving/#received-payment'| relative_url }}).

If the user does not have a channel open, or one with enough inbound liquidity, a new channel needs to be opened. Any additional fees required to open a channel need to be communicated to users when [creating a payment request]({{ '/guide/daily-spending-wallet/requesting/#fees'| relative_url }}), and before they share it.

In this wallet channel opens and management are done by a lightning service provider (LSP). Channel opens happen [on-demand]({{ '/guide/how-it-works/lightning-services/#on-demand-liquidity'| relative_url }}) and use [zero confirmation channel opens]({{ '/guide/how-it-works/lightning-services/#zero-confirmation'| relative_url }}) so users can receive and spend their bitcoin instantly.

## Receiving on-chain payments

> This wallet maintains a single lightning balance. To do this but still allow on-chain sending and requesting, a LWS is used that conducts [submarine swaps](https://blog.muun.com/a-closer-look-at-submarine-swaps-in-the-lightning-network/) between lightning and on-chain.

Reviewer 2: ```suggestion
This wallet maintains a single lightning balance. To do this, while still allowing for on-chain sending and requesting, a LWS is used that conducts [submarine swaps](https://blog.muun.com/a-closer-look-at-submarine-swaps-in-the-lightning-network/) between lightning and on-chain.
```

{% include tip/open.html color="blue" icon="info" label="Experimental options" %}

[Peerswap](https://www.peerswap.dev/) or [collaborative funding](https://bitcoinops.org/en/topics/dual-funding/) may be more trustless options that do not require a third-party LWS to receive and send on-chain payments. However, these are not currently widely adopted.

{% include tip/close.html %}

### Failed swaps

In the vast majority of scenarios, the submarine swap succeeds and the user is able to receive on-chain bitcoin to their lightning wallet without even thinking about this complex process happening behind the scenes. However, in some edge cases, the swap can fail for various reasons.

- LWS swap service is offline
- Receiving more than the users inbound capacity
- Re-using a swap-address to receive another on-chain payment

> When this occurs, the funds will show up as "pending" in the user's activity and give them a prompt on the home page that action needs to be taken. The user is given two options to deal with this. Either re-trying the swap or sending to an external on-chain address.

Reviewer 2: ```suggestion
When this occurs, the funds will show up as "pending" in the user's activity, and a prompt on the home screen appears that lets the user know that an action needs to be taken. They are given two options to deal with this, either re-trying the swap or sending to an external on-chain address.
```

> If the swap fails again, the wallet can open a new channel locally without using the LWS. The pending message can be the same as the user doesn't need to know the complexities of what is happening behind the scenes.

Reviewer 2: ```suggestion
If the swap fails again, the wallet can open a new channel locally without using the LWS. The pending message can be the same, as the user doesn't need to know the complexities of what is happening behind the scenes.
```
